.github/scripts/gen_historical_disabled_issue_data.py

```diff
@@ -0,0 +1,112 @@
+import json
+import os
+import subprocess
+from concurrent.futures import as_completed, ThreadPoolExecutor
+from datetime import datetime, timezone
+from tempfile import TemporaryDirectory
+from typing import List
+
+
+BRANCH = "generated-stats"
+FILE = "stats/disabled-tests-condensed.json"
+
+
+def get_info(commit: str, day: str, timestamp: int) -> List[dict]:
+    contents = (
+        subprocess.check_output(
+            ["git", "show", f"{commit}:{FILE}"],
+        )
+        .decode("utf-8")
+        .strip()
+    )
+    json_data = json.loads(contents)
+    return format_info(json_data, day, timestamp)
+
+
+def format_info(info: dict, day: str, timestamp: int) -> List[dict]:
+    as_array = []
+    for name, item in info.items():
+        issue_number, _, platforms = item
+        as_array.append(
+            {
+                "day": day,
+                "timestamp": timestamp,
+                "name": name,
+                "issueNumber": issue_number,
+                "platforms": platforms,
+            }
+        )
+    return as_array
+
+
+def gen_all_history():
+    with TemporaryDirectory() as tmpdir:
```

Comment:
I think you could consider redo this using `GitPython` instead of using subprocess for a more concise script, i.e. https://github.com/pytorch/pytorch-integration-testing/blob/main/vllm-benchmarks/upload_benchmark_results.py#L90-L100.  But it's a minor point